test/conformance/api/v1beta1/service_account_test.go

```diff
@@ -0,0 +1,58 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"strings"
+	"testing"
+
+	. "knative.dev/serving/pkg/testing/v1beta1"
+	"knative.dev/serving/test"
+	v1b1test "knative.dev/serving/test/v1beta1"
+)
+
+const (
+	invalidServiceAccountName = "foo@bar.baz"
+)
+
+func TestServiceAccountValidation(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.PizzaPlanet1,
+	}
+
+	defer test.TearDown(clients, names)
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+
+	t.Logf("Creating a new Service %s", names.Service)
+	service := v1b1test.Service(names, WithServiceAccountName(invalidServiceAccountName))
+	v1b1test.LogResourceObject(t, v1b1test.ResourceObjects{Service: service})
+
+	_, err := clients.ServingBetaClient.Services.Create(service)
+	if err == nil {
+		t.Fatal("Expected Service creation to fail")
+	}
+
+	if !strings.Contains(err.Error(), "serviceAccountName: spec.template.spec."+invalidServiceAccountName) {
```

Comment:
Just a nit: but I think there should be a single error and we should be able to check against it?

```
if got, want := err.Error(), "serviceAccountName: spec.template.spec."+invalidServiceAccountName; got != want {
  t.Errorf("err = %s, want: %s, diff:\n%s", got, want, cmp.Diff(got, want)
}
```

Reply:
In this case `want` is a substring of the error msg.  Are you suggesting something like below?

```
if got, want := err.Error(), "serviceAccountName: spec.template.spec."+invalidServiceAccountName; !strings.Contains(got, want) {
  t.Errorf("got = %s, want: %s ", got, want)
}
```

Reply:
Eh, if we can't check the whole error? Could you post it here? That would work as well.

Reply:
I was hesitant to check for exact error because that seems like a very strict error check for conformance test.

Reply:
Well, conformance is about things conforming to something :-)
But it's fine. Substring should work.
Let me know when you push the `if got, want...` change.

Reply:
@vagababov : I have updated the PR